networking/p2p/src/main/java/tech/pegasys/teku/networking/p2p/connection/ConnectionManager.java

```diff
@@ -91,17 +94,40 @@ protected SafeFuture<?> doStart() {
     synchronized (this) {
       staticPeers.forEach(this::createPersistentConnection);
     }
-    periodicPeerSearch =
-        asyncRunner.runWithFixedDelay(
-            this::searchForPeers,
-            DISCOVERY_INTERVAL,
-            error -> LOG.error("Error while searching for peers", error));
-    connectToBestPeers(emptyList());
-    searchForPeers();
+    startSearchPeerTask();
     peerConnectedSubscriptionId = network.subscribeConnect(this::onPeerConnected);
     return SafeFuture.COMPLETE;
   }
 
+  @SuppressWarnings("FutureReturnValueIgnored")
+  private void startSearchPeerTask() {
+    this.periodicPeerSearch =
+        asyncRunner.runCancellableAfterDelay(
+            () -> searchForPeers().thenPeek(__ -> createFollowupSearchPeerTask()), Duration.ZERO);
```

Comment:
Not sure why this is using `thenPeek`. It means the `createFollowupSearchPeerTask` won't execute of the future from `searchForPeers` completes exceptionally which would then mean discovery halts entirely.

Also, it's very weird to be saying run after delay but having a zero delay. Given we only cancel on shutdown and the cancel will only work if the task hasn't already started executing, I suspect we should just leave `periodicPeerSearch` as `null` here and call `searchForPeers()` without going through the async executor at all.

So it winds up something like:
```java
searchForPeers()
        .alwaysRun(this::createFollowupSearchPeerTask)
        .finish(error -> LOG.error("Error while searching for peers", error));
```